### ISSUE TITLE
Segfault when trying to add a mixer or monitor

### ISSUE THREAD

**Issue by Natronium** [OpenMusicKontrollers/patchmatrix#23]
trying to add a mixer or monitor of any size from the canvas' right click menu causes patchmatrix to crash with a segfault. this happens both with the latest version from git (79cac7) that i built myself and the v0.24 pre-built binaries.

i'm running on amd64 Debian Testing with X and Pipewire using Intel HD graphics, and the output from patchmatrix on stderr is
```
$ ./patchmatrix
./patchmatrix 0.25.5
Copyright (c) 2016-2021 Hanspeter Portner (dev@open-music-kontrollers.ch)
Released under Artistic License 2.0 by Open Music Kontrollers
[W][000281488.478214][pipewire-jack.c:4699 jack_set_session_callback()] jack-client 0x55c733df5590: not implemented
[E][000281488.478772][pipewire-jack.c:656 on_error()] jack-client 0x55c733df5590: error id:4 seq:11 res:-2 (No such file or directory): set property error for id -1: No such file or directory
[E][000281488.478814][pipewire-jack.c:656 on_error()] jack-client 0x55c733df5590: error id:4 seq:12 res:-2 (No such file or directory): set property error for id -1: No such file or directory
```
everything mostly works fine, as far as i can tell, aside from the part where it segfaults (and the icon rendering issue mentioned in #22). i don't think the jack/pipewire complaints are related to the segfault since the backtrace from the coredump seems to be pointing to pugl doing something that makes X very sad or vice versa:

```
Core was generated by `./bin/patchmatrix'.
Program terminated with signal SIGSEGV, Segmentation fault.
#0  0x00007f6d7f445000 in ?? () from /lib/x86_64-linux-gnu/libxcb-glx.so.0
[Current thread is 1 (Thread 0x7f6d7f7cbb80 (LWP 1200962))]
(gdb) bt
#0  0x00007f6d7f445000 in ?? () from /lib/x86_64-linux-gnu/libxcb-glx.so.0
#1  0x00007f6d7fdb6262 in XGetErrorText () from /lib/x86_64-linux-gnu/libX11.so.6
#2  0x00007f6d7fdd6b0b in ?? () from /lib/x86_64-linux-gnu/libX11.so.6
#3  0x00007f6d7fdd8473 in _XDefaultError () from /lib/x86_64-linux-gnu/libX11.so.6
#4  0x00007f6d7fdd85d4 in _XError () from /lib/x86_64-linux-gnu/libX11.so.6
#5  0x00007f6d7fdd5097 in ?? () from /lib/x86_64-linux-gnu/libX11.so.6
#6  0x00007f6d7fdd5135 in ?? () from /lib/x86_64-linux-gnu/libX11.so.6
#7  0x00007f6d7fdd5d6a in _XEventsQueued () from /lib/x86_64-linux-gnu/libX11.so.6
#8  0x00007f6d7fdb75ea in XFlush () from /lib/x86_64-linux-gnu/libX11.so.6
#9  0x000055bd425cfc9a in puglDispatchX11Events ()
#10 0x000055bd425d1d43 in puglUpdate ()
#11 0x000055bd425640bc in main ()
```
that's about the extent of my gdb know-how, otherwise i'd've debugged this a little further for you. let me know if you want me to extract any other information/perform any experiments for you. i've uploaded the core dump (from the x86_64 bullseye build of v0.24.0) for your perusal as well.

[patchmatrix-0.24.0-x86_64-linux-gnu-bullseye-coredump.zip](https://github.com/OpenMusicKontrollers/patchmatrix/files/6479364/patchmatrix-0.24.0-x86_64-linux-gnu-bullseye-coredump.zip)

**Comment by ventosus** [comment]
You likely have not installed the (hand-built) application and the mixer/monitor binaries (yes, they are separate binaries) are not found because they are not in your PATH.

**Comment by Natronium** [comment]
yep! that does it. adding the bin directory to my PATH fixes the problem. FWIW this was happening with the prebuilt binaries as well, not just the hand built ones (since in both cases the helper binaries weren't on the PATH. I noticed the binaries, but figured it would look around or provide some error output rather than just... segfaulting

**Comment by ventosus** [comment]
thanks for reporting

segfault is fixed in 56e7230c96d169de775395cba272ceb19e888b21

there will be no error reporting, as anybody can start a mixer/monitor binary, e.g. they can be starten from *this* patchmatrix ui, *another* patchmatrix ui, from the console, etc.
